### PR TITLE
closes #816 Disable email catcher popup

### LINKED ISSUE #816: Disable email catcher popup
MailChimp now provides a popup so this one is not necessary.

### PR DESCRIPTION
**Pull request by ebizmarts-keller**
- Comment entry in config file to leave possibility to re-enable easily from the code
- Disable popup from upgrade script

### CODE REVIEW

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 1]
```diff
@@ -323,7 +323,7 @@
                         </order_batch_amount>
                     </fields>
                 </ecommerce>
-                <emailcatcher>
+                <!--<emailcatcher>
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: remove code if not needed

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 2]
```diff
@@ -472,7 +472,7 @@
                             </depends>
                         </popup_insist>
                     </fields>
-                </emailcatcher>
+                </emailcatcher>-->
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: remove code if not needed

File: app/code/community/Ebizmarts/MailChimp/sql/mailchimp_setup/mysql4-upgrade-1.1.12-1.1.13.php [hunk 3]
```diff
@@ -60,4 +60,7 @@
 ;
 $attribute->save();
 
+Mage::helper('mailchimp')->saveMailChimpConfig(array(array(Ebizmarts_MailChimp_Model_Config::ENABLE_POPUP, 0)), 0, 'default');
+
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/sql/mailchimp_setup/mysql4-upgrade-1.1.12-1.1.13.php]: remove unnecesary lines between code

File: app/code/community/Ebizmarts/MailChimp/sql/mailchimp_setup/mysql4-upgrade-1.1.12-1.1.13.php [hunk 4]
```diff
@@ -60,4 +60,7 @@
 ;
 $attribute->save();
 
+Mage::helper('mailchimp')->saveMailChimpConfig(array(array(Ebizmarts_MailChimp_Model_Config::ENABLE_POPUP, 0)), 0, 'default');
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/sql/mailchimp_setup/mysql4-upgrade-1.1.12-1.1.13.php]: magic numbers, change `0` to var or constant.
also, I think `default` can be read from a magento constant in a core class

Comment (ebizmarts-keller) [reply app/code/community/Ebizmarts/MailChimp/sql/mailchimp_setup/mysql4-upgrade-1.1.12-1.1.13.php]: Thanks, I changed the way the pop is disabled, I realized using 'default' doesn't take into consideration if the config exists in the website or store scopes, now it deletes that configuration for all scopes.

File: app/code/community/Ebizmarts/MailChimp/sql/mailchimp_setup/mysql4-upgrade-1.1.12-1.1.13.php [hunk 5]
```diff
@@ -60,7 +60,6 @@
 ;
 $attribute->save();
 
-Mage::helper('mailchimp')->saveMailChimpConfig(array(array(Ebizmarts_MailChimp_Model_Config::ENABLE_POPUP, 0)), 0, 'default');
-
+$installer->deleteConfigData(Ebizmarts_MailChimp_Model_Config::ENABLE_POPUP);
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/sql/mailchimp_setup/mysql4-upgrade-1.1.12-1.1.13.php]: are we sure `deleteConfigData` performs the same on all Magento 1 versions we support?

**Pull request merged**